### PR TITLE
Added test cases for the check ec2_imdsv2_enabled

### PR DESCRIPTION
**Pull request by Ninad-Lunge-Comprinno**
### Context

Adding test coverage for the EC2 IMDSv2 enforcement check.

### Description

This change adds a comprehensive test suite for the `ec2_imdsv2_enabled` check. It verifies the following scenarios:

* Instances with IMDSv2 fully enforced
* Instances that do not enforce IMDSv2
* Instances missing metadata options
* No running EC2 instances
* Error handling for AWS client failures

These tests ensure the check behaves as expected across different edge cases.

### Checklist

* [x] Added new checks? If yes, reviewed necessary permissions
* [x] Code covered by tests
* [x] Documentation follows [[Google Python Style Guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings)](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings)
* [x] Considered if backporting is needed

### License

I confirm that my contribution is made under the terms of the **Apache 2.0 license**.

### CODE REVIEW

File: library/aws/tests/ec2/test_ec2_imdsv2_enabled.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        """Test ClientError is handled properly."""
+        self.mock_client.get_paginator.side_effect = ClientError(
+            {"Error": {"Code": "UnauthorizedOperation", "Message": "You are not authorized"}}, "DescribeInstances"
+        )
+
+        report = self.check.execute(self.mock_session)
+        assert report.status == CheckStatus.UNKNOWN
+        assert len(report.resource_ids_status) == 1
+        assert report.resource_ids_status[0].status == CheckStatus.UNKNOWN
+        assert "Error retrieving EC2 metadata service configuration" in (report.resource_ids_status[0].summary or "")
```
[the remark this answers is not in the document]
Comment (Ninad-Lunge-Comprinno) [reply library/aws/tests/ec2/test_ec2_imdsv2_enabled.py:166]: @aafaq-rashid-comprinno Please review the updated code

**Review by aafaq-rashid-comprinno — APPROVED** [review]
